internal/controller/promotion/argocd.go

```diff
@@ -510,12 +537,19 @@ func applyArgoCDSourceUpdate(
 	newFreight kargoapi.FreightReference,
 	update kargoapi.ArgoCDSourceUpdate,
 ) (argocd.ApplicationSource, error) {
+
+	ctx := context.Background()
+	logger := logging.LoggerFromContext(ctx)
+	updateLogger := logger.WithValues("source", source, "update", update)
+	updateLogger.Debug("About to apply ArgoCD source update")
+
 	if source.Chart != "" || update.Chart != "" {
 		// Infer that we're dealing with a chart repo. No need to normalize the
 		// repo URL here.
 
 		// Kargo uses the "oci://" prefix, but Argo CD does not.
 		if source.RepoURL != strings.TrimPrefix(update.RepoURL, "oci://") || source.Chart != update.Chart {
+			updateLogger.Debug("Kargo uses oci:// prefix, but Argo CD does not, won't update.")
```

Comment:
This is not factually correct. The `TrimPrefix` is to make the comparison of a Kargo value against an Argo CD value works (by ensuring we compare the two URLs _without_ the `oci://` prefix). This failing does, however, not mean what is described in the log message — it simply means the update does not apply to this source.

Reply:
Ah, that makes sense. Any recommendation as to what this message should be then?

Reply:
Something like

> Source update not applicable to Application source, skipping

maybe? But I do wonder if logging this at debug level does not cause immense verbosity, as it is common for the mixed bag of source updates to only apply to a subset. Which means `Trace` might be a better level.